src/format/opamFile.ml

```diff
@@ -2664,13 +2691,44 @@ module OPAMSyntax = struct
            else None
          in
          let t = { t with metadata_dir } in
+         let t =
+           match OpamStd.String.Map.find_opt subpath_xfield t.extensions with
+           | Some (_, String (pos,subpath)) ->
+             let extensions =
+               OpamStd.String.Map.remove subpath_xfield t.extensions
+             in
+             let url =
+               OpamStd.Option.Op.(t.url >>| (fun u ->
+                   (match u.subpath with
+                    | Some sb ->
+                      Pp.warn ~pos "%s already defined (%s) in the opam file, \
+                                    it is replaced by %s."
+                        (OpamConsole.colorise `underline "subpath") sb subpath
+                    | None -> ());
+                   URL.with_subpath subpath u))
+             in
+              with_opam2_1_restriction { t with url; extensions }
+           | Some (pos, _) ->
+             Pp.warn ~pos ~strict:true "%s field contains a non string value"
+               (OpamConsole.colorise `underline subpath_xfield);
+             t
+           | None -> t
+         in
          match OpamPackage.of_filename filename with
          | Some nv -> with_nv nv t
          | None -> t)
       (fun (filename, t) ->
          filename,
+         let t =
+           match t.url with
+           | Some ({ URL.subpath = Some sb ; _ } as url) ->
+             add_extension t subpath_xfield (String (pos_null, sb))
```

Comment:
This could also be made conditional depending on the `opam-version:` of the file being written. Of course the default should remain `2.0` when creating files.

Reply:
Same as upper comment, here, it is the subpath field that is concerned, and added with the availability restrictions. The generated file is still 2.0 and on next opam file version we won't need this `pp` workaround